### PR TITLE
Add static node settings to set default values for max merged segment sizes

### PR DESCRIPTION
**Pull request by tlrx**
This change introduces 2 static node-level settings that allow to override the default values for the maximum merged segment sizes in tiered and time-based merge policies.

New default values can be defined like this in `elasticsearch.yml`:

```yaml
indices:
  merge:
    policy:
      max_merged_segment: "5gb"
      max_time_based_merged_segment: "25gb"
```

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java [hunk 1]
```diff
@@ -111,17 +111,33 @@ public final class MergePolicyConfig {
...
     private volatile Type mergePolicyType;
+    private final ByteSizeValue defaultMaxMergedSegment;
+    private final ByteSizeValue defaultMaxTimeBasedMergedSegment;
 
     public static final double DEFAULT_EXPUNGE_DELETES_ALLOWED = 10d;
     public static final ByteSizeValue DEFAULT_FLOOR_SEGMENT = new ByteSizeValue(2, ByteSizeUnit.MB);
     public static final int DEFAULT_MAX_MERGE_AT_ONCE = 10;
     public static final ByteSizeValue DEFAULT_MAX_MERGED_SEGMENT = new ByteSizeValue(5, ByteSizeUnit.GB);
+    public static final Setting<ByteSizeValue> DEFAULT_MAX_MERGED_SEGMENT_SETTING = Setting.byteSizeSetting(
+        "merge.policy.default.max_merged_segment",
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java:122]: I'd prefer something like:

```suggestion
        "indices.merge.policy.max_merged_segment",
```

though adding `default` could also be fine by me: `indices.default.merge.policy.max_merged_segment`.

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java (hunk 1 above)
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java]: I pushed 63a93ecc0a8a3ce0aac91ac327d75da940105748 to use `indices.merge.policy.max_merged_segment`

**Comment by tlrx** [comment]
Thanks Henning!

**Pull request merged**